src/postamble.js

```diff
@@ -225,7 +225,7 @@ function run(args) {
   if (preloadStartTime === null) preloadStartTime = Date.now();
 
   if (runDependencies > 0) {
-#if ASSERTIONS
+#if ASSERTIONS && !BINARYEN_ASYNC_COMPILATION
```

Comment:
We do enable assertions in -O0 builds (which should not be used in production). I guess the bot tests -O0?

In any case, we can probably put this behind a different flag. No existing flag seems suitable, so maybe we should just add one for runtime logging, `RUNTIME_LOGGING` (just need to add it in `src/settings.js`, then can use it).